sdk/python/kfp/client/client_test.py

```diff
@@ -41,5 +44,82 @@ def test_invalid(self, name: str):
+    'compiler', 'test_data', 'pipelines')
+
+PIPELINE_TEST_CASES = [
+    'pipeline_with_importer',
+    'pipeline_with_ontology',
+    'pipeline_with_if_placeholder',
+    'pipeline_with_concat_placeholder',
+    'pipeline_with_resource_spec',
+    'pipeline_with_various_io_types',
+    'pipeline_with_reused_component',
+    'pipeline_with_after',
+    'pipeline_with_condition',
+    'pipeline_with_nested_conditions',
+    'pipeline_with_nested_conditions_yaml',
+    'pipeline_with_loops',
+    'pipeline_with_nested_loops',
+    'pipeline_with_loops_and_conditions',
+    'pipeline_with_params_containing_format',
+    'lightweight_python_functions_pipeline',
+    'lightweight_python_functions_with_outputs',
+    'xgboost_sample_pipeline',
+    'pipeline_with_metrics_outputs',
+    'pipeline_with_exit_handler',
+    'pipeline_with_env',
+    'component_with_optional_inputs',
+    'pipeline_with_gcpc_types',
+    'pipeline_with_placeholders',
+    'pipeline_with_task_final_status',
+    'pipeline_with_task_final_status_yaml',
+    'component_with_pip_index_urls',
+]
+
+
+class TestOverrideCachingOptions(parameterized.TestCase):
+
+    @parameterized.parameters(PIPELINE_TEST_CASES)
+    def test_override_caching_from_yaml(self, pipeline_base_name: str):
+        pipeline_path = os.path.join(PIPELINES_TEST_DATA_DIR,
+                                     f'{pipeline_base_name}.yaml')
+        with open(pipeline_path) as f:
+            yaml_dict = yaml.safe_load(f)
+            test_client = client.Client(namespace='dummy_namespace')
+            test_client._override_caching_options(yaml_dict, False)
+            for _, task in yaml_dict['root']['dag']['tasks'].items():
+                assert task['cachingOptions']['enableCache'] == False
+
+    def test_override_caching_from_pipeline(self):
+        from kfp.v2.compiler import Compiler
+        from kfp.v2.dsl import component
+        from kfp.v2.dsl import pipeline
+
+        @component
+        def hello_world(text: str) -> str:
+            """Hello world component."""
+            return text
+
+        @pipeline(name='hello-world', description='A simple intro pipeline')
+        def pipeline_hello_world(text: str = 'hi there'):
+            """Hello world pipeline."""
+
+            hello_world(text=text).set_caching_options(False)
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            temp_filepath = os.path.join(tempdir, 'hello_world_pipeline.yaml')
+            Compiler().compile(
+                pipeline_func=pipeline_hello_world, package_path=temp_filepath)
+
+            with open(temp_filepath, 'r') as f:
+                pipeline_obj = yaml.load(f)
+                test_client = client.Client(namespace='dummy_namespace')
+                test_client._override_caching_options(pipeline_obj, True)
+                for _, task in pipeline_obj['root']['dag']['tasks'].items():
+                    assert task['cachingOptions']['enableCache'] == True
```

Comment:
Note: we've only recently moved toward using `pytest` as our unit test executor (as opposed to `unittest`), since it has better test discovery than `unittest`. This is why we still use `self.assertEqual()` (supported by both `unittest` and `pytest`) instead of bare `assert` (only supported by `pytest`).

What do you think about using `self.assertEqual()` here for consistency with the way other unit tests are done that way this doesn't appear like a special/unusual case, then we can migrate in one larger effort in the future? As opposed to an incremental migration?

I could go either way on this -- curious what you think.

Reply:
The PR that switches to using `pytest`: https://github.com/kubeflow/pipelines/pull/7869